### ISSUE TITLE
correctly parse commits which lack some of the expected metadata

### ISSUE THREAD

**Issue by jixiuf** [magit/magit#1919]
[next]
```
961958f * mysql-5.6.25 origin/5.6 
b8e5508 *   Empty version change upmerge
        |\  
f5ac718 | * Raise version number after cloning 5.5.44
c61fdfb * | 
f5760e7 * | 
6dd6b53 * | 
d9b1ac0 * | Bug #20857652 : SERVER CRASHES WITH CREATE/DROP USER                 STATEMENT WITH A PARTICULAR SEQUENCE

```
when I press Enter on c61fdfb I got this
    magit-diff-wash-revision: Search failed: "^\\(\\(---\\)\\| .\\)"

git log
```
commit 961958f355d93f9d015de818e9ea1a6eeb1029d2
Author: mysql-builder@oracle.com <>
Date:   2015-05-05 12:13:56 +0200

commit b8e5508b4981db85e542d3e11c7e877eb421e621
Merge: c61fdfb f5ac718
Author: Balasubramanian Kandasamy <balasubramanian.kandasamy@oracle.com>
Date:   2015-05-04 12:09:18 +0200

    Empty version change upmerge

commit f5ac718b1db7a012d113b051ef74932c6e0a29df
Author: Balasubramanian Kandasamy <balasubramanian.kandasamy@oracle.com>
Date:   2015-05-04 12:06:52 +0200

    Raise version number after cloning 5.5.44

commit c61fdfb80562155d6fe08ccf76ac65d299c14817
Author: mysql-builder@oracle.com <>
Date:   2015-05-01 12:39:55 +0200

commit f5760e771ac5e15b87738dd275b0aa63c12aebc7
Author: mysql-builder@oracle.com <>
Date:   2015-05-01 07:15:24 +0530
```

**Comment by tarsius** [comment]
This seems to be a very bad commit (no message, email as name, no email). Was Git used to create this commit or some importer script?

**Comment by jixiuf** [comment]
I guess should be  script .
I am just mysql git source  and got this error .

**Comment by tarsius** [comment]
I will look into this after the release but cannot promise anything. The regular expressions used to parse diffs (and other things) should be improved, for example it would probably be a good idea to use non-printing characters as field separators (e.g. the field separator character :-). This might be enough to fix this issue. Or it might not and then it gets more complicated: the regexps could be relaxed, but then the risk increases that "correct" commits, which feature all of the expected metadata, but whose commit messages are somehow "unusual" would no longer be parsed correctly. We will see.

**Comment by tarsius** [comment]
This has been fixed in #2216 and will be released in v2.3.0.
